javaparser-core/src/main/java/com/github/javaparser/ParseStart.java

```diff
@@ -8,21 +8,18 @@
 import com.github.javaparser.ast.stmt.BlockStmt;
 import com.github.javaparser.ast.stmt.Statement;
 
-import java.util.List;
-
 /**
  * The start production for JavaParser.
  * Tells JavaParser what piece of Java code it can expect.
  * For example,
  * COMPILATION_UNIT indicates a complete Java file,
  * and CLASS_BODY would indicate the part of a class that is within { and }.
- * @see JavaParser#parseStructure(ParseStart, Provider)
+ * @see JavaParser#parse(ParseStart, Provider)
  */
 @FunctionalInterface
 public interface ParseStart<R> {
 	ParseStart<CompilationUnit> COMPILATION_UNIT = ASTParser::CompilationUnit;
 	ParseStart<BlockStmt> BLOCK = ASTParser::Block;
-	ParseStart<List<Statement>> STATEMENTS = ASTParser::Statements;
```

Comment:
Because JavaParser now always inserts comments in whatever it has parsed. The only requirement is that it has to be a Node. A List is not a Node. So, just for this I would have had to create a few separate methods that avoid adding comments, or maybe iterate the list and insert comments into each of them... Lots of trouble, and easily avoided by using BLOCK, for example.

This does make a case for a class that is a Node *and* a List. This would also help a lot with our mutable list parenting issues.